packages/playwright-core/src/server/injected/roleUtils.ts

```diff
@@ -461,6 +461,35 @@ export function getElementAccessibleDescription(element: Element, includeHidden:
   return accessibleDescription;
 }
 
+export function getElementAccessibleErrorMessage(element: Element): string {
+  const cache = cacheAccessibleErrorMessage;
+  let accessibleErrorMessage = cacheAccessibleErrorMessage?.get(element);
+
+  if (accessibleErrorMessage === undefined) {
+    accessibleErrorMessage = '';
+
+    const ariaInvalid = element.getAttribute('aria-invalid');
+    const isAriaInvalid = ariaInvalid !== null && ariaInvalid.toLowerCase() !== 'false';
+    if (isAriaInvalid) {
+      const errorMessageId = element.getAttribute('aria-errormessage');
+      if (errorMessageId) {
+        const errorMessages = getIdRefs(element, errorMessageId);
+        if (errorMessages.length) {
+          const parts = errorMessages.map(errorMessage => asFlatString(
+              getTextAlternativeInternal(errorMessage, {
+                visitedElements: new Set(),
+                embeddedInDescribedBy: { element: errorMessage, hidden: isElementHiddenForAria(errorMessage) },
```

Comment:
Looking at [this spec](https://w3c.github.io/core-aam/#ariaErrorMessage), I think this should work. Ideally we'd have a separate `embeddedInErrorMessage`, but for now let's just add a comment:

```
Ideally, this should be a separate "embeddedInErrorMessage", but it would follow the exact same rules.
Relevant vague spec: https://w3c.github.io/core-aam/#ariaErrorMessage.
```